spec/requests/hyku_addons/account_settings_request_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe "::HykuAddons::AccountSettingsController", type: :request do
+  let(:user) { FactoryBot.create(:admin) }
+  let!(:account) { create(:account) }
+  before do
+    login_as(user, scope: :user)
+    allow(Apartment::Tenant).to receive(:switch).with(account.tenant) do |&block|
+      block.call
+    end
+    account.switch!
+  end
+
+  after do
+    account.reset!
+  end
+
+  describe 'GET /admin/account_settings' do
+    it "GETS index  /admin/account_settings" do
+      get admin_account_settings_url
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe "Update single keys in settings hash" do
+    it "can update a single settings key eg  contact_email" do
+      patch update_single_admin_account_setting_url(id: account, partial_name: 'contact_email'), params: { "account" => { "settings" => { "contact_email" => "ab@abc.com" } } }
```

Comment:
Why do you think so, seeing that we use it here in the controller action

            format.js { render template: "/hyku_addons/account_settings/settings/#{params[:partial_name]}.js.erb" }

and that is because for each key in the settings jsonb we have a  a partial  that we use do an ajax update when updating just a single key from the settings hash.

Reply:
The request here should be routing to update_single which doesn't appear to use the `partial_name` param.  Only JS requests to the `edit` action use the param as far as I can tell.